### ISSUE TITLE
plotting multiple graph routes error

### ISSUE THREAD

**Issue by MegaCreater** [gboeing/osmnx#221]
import osmnx as ox, networkx as nx, matplotlib.pyplot as plt, math as m
from haversine import haversine

org = (27.5075, 77.6779)
dst = (27.5067, 77.6694)
dist_fl = (int(haversine(org, dst)*1000))*2
G = ox.graph_from_point(org, simplify=False, distance=dist_fl)
orig_node = ox.get_nearest_node(G, org, method='euclidean')
target_node = ox.get_nearest_node(G, dst, method='euclidean')

nodes, edges = ox.graph_to_gdfs(G)
streets_per_node = ox.count_streets_per_node(G)
route = nx.shortest_path(G, source=orig_node,  target=target_node, weight='length', method='dijkstra')
node_route = nodes.loc[route]

route_list = [[1352186296, 1352186345], [1352186345, 1352186150, 1350260490, 1352186112, 1352186349, 1350260516, 4753659536L, 4753498382L], [4753498382, 4753498383L, 4753498390L], [4753498390, 4753498391L, 4753498392L, 4753498393L, 4753498394L], [4753498394, 4753498395L, 4753498396L, 4753498397L, 4753498398L], [4753498398, 4753478108L], [4753478108, 4753498399L, 4753498400L, 4753498401L], [4753498401, 4753498402L, 4753498403L], [4753498403, 4753498404L, 4753498405L, 4753657826L, 4753498406L, 4753498407L, 4753498408L, 4753498409L], [4753498409, 4753498410L, 4753498411L, 4753498412L], [4753498412, 3803339674L], [3803339674, 1352186110, 1352186253, 2265692728L], [2265692728, 1352186308, 1352186133, 1352186328, 1352186316, 1352186071, 1352186187, 1352186095], [1352186095, 1352186361, 1352186263]]

coord_list =[(77.6776527, 27.5069438), (77.6775911, 27.5061476), (77.6747421, 27.5061048), (77.6747447, 27.5065153), (77.6742216, 27.5065135), (77.6741438, 27.507339), (77.6740672, 27.5073159), (77.6738192, 27.5079793), (77.6736687, 27.5079667), (77.6726402, 27.5084229), (77.6720677, 27.5081814), (77.6717673, 27.5093129), (77.6704672, 27.5094812), (77.6690214, 27.507629)]

color_list = ['green', 'green', 'green', 'green', 'red', 'green', 'green', 'green', 'green', 'green', 'green', 'green', 'red', 'red']

fig, ax = ox.plot_graph_routes(G, route_list, fig_height=10, fig_width=10, save=True, filename='outfile_data', show=False, close=False, axis_off=False, edge_linewidth=1, node_size=10, route_color = color_list)
plt.show()
![capture2](https://user-images.githubusercontent.com/45305845/49062785-4d611380-f23b-11e8-84ff-dcb8c85c38f2.JPG)

**Comment by MegaCreater** [comment]
'''but if we plot each route on different map the result is perfect. but i want to plot multiple route on single map.'''
import osmnx as ox, networkx as nx, matplotlib.pyplot as plt, math as m
from haversine import haversine

org = (27.5075, 77.6779)
dst = (27.5067, 77.6694)
dist_fl = (int(haversine(org, dst)*1000))*2
G = ox.graph_from_point(org, simplify=False, distance=dist_fl)
orig_node = ox.get_nearest_node(G, org, method='euclidean')
target_node = ox.get_nearest_node(G, dst, method='euclidean')

nodes, edges = ox.graph_to_gdfs(G)
streets_per_node = ox.count_streets_per_node(G)
route = nx.shortest_path(G, source=orig_node,  target=target_node, weight='length', method='dijkstra')
node_route = nodes.loc[route]

route_list = [[1352186296, 1352186345], [1352186345, 1352186150, 1350260490, 1352186112, 1352186349, 1350260516, 4753659536L, 4753498382L], [4753498382, 4753498383L, 4753498390L], [4753498390, 4753498391L, 4753498392L, 4753498393L, 4753498394L], [4753498394, 4753498395L, 4753498396L, 4753498397L, 4753498398L], [4753498398, 4753478108L], [4753478108, 4753498399L, 4753498400L, 4753498401L], [4753498401, 4753498402L, 4753498403L], [4753498403, 4753498404L, 4753498405L, 4753657826L, 4753498406L, 4753498407L, 4753498408L, 4753498409L], [4753498409, 4753498410L, 4753498411L, 4753498412L], [4753498412, 3803339674L], [3803339674, 1352186110, 1352186253, 2265692728L], [2265692728, 1352186308, 1352186133, 1352186328, 1352186316, 1352186071, 1352186187, 1352186095], [1352186095, 1352186361, 1352186263]]

coord_list =[(77.6776527, 27.5069438), (77.6775911, 27.5061476), (77.6747421, 27.5061048), (77.6747447, 27.5065153), (77.6742216, 27.5065135), (77.6741438, 27.507339), (77.6740672, 27.5073159), (77.6738192, 27.5079793), (77.6736687, 27.5079667), (77.6726402, 27.5084229), (77.6720677, 27.5081814), (77.6717673, 27.5093129), (77.6704672, 27.5094812), (77.6690214, 27.507629)]

color_list = ['green', 'green', 'green', 'green', 'red', 'green', 'green', 'green', 'green', 'green', 'green', 'green', 'red', 'red']


for x in range (len(route_list)):
    fig, ax = ox.plot_graph_route(G, route_list[x], fig_height=10, fig_width=10, save=True, filename='outfile_data', show=False, close=False, axis_off=False, edge_linewidth=1, node_size=10, route_color = color_list[x])
    plt.show()

**Comment by gboeing** [comment]
It's not clear to me what I'm seeing here. Would you mind providing some more narrative detail of what's going on and what you're intending, for example using the issue template?

**Comment by gboeing** [comment]
@MegaCreater 

```python
import networkx as nx
import osmnx as ox
ox.config(log_console=True, use_cache=True)

G = ox.graph_from_place('Piedmont, CA, USA', network_type='drive')

# pick 4 random nodes as origins/destinations for the 2 routes
orig1 = list(G.nodes())[0]
dest1 = list(G.nodes())[-1]
orig2 = list(G.nodes())[50]
dest2 = list(G.nodes())[-50]

# calculate shortest paths for the 2 routes
route1 = nx.shortest_path(G, orig1, dest1, weight='length')
route2 = nx.shortest_path(G, orig2, dest2, weight='length')

# create route colors
rc1 = ['r'] * (len(route1) - 1)
rc2 = ['b'] * len(route2)
rc = rc1 + rc2
nc = ['r', 'r', 'b', 'b']

# plot the routes
fig, ax = ox.plot_graph_routes(G, [route1, route2], route_color=rc, orig_dest_node_color=nc, node_size=0)
```
![osmnx plot multiple routes different colors](https://user-images.githubusercontent.com/4977197/49740360-f8bd8f80-fc61-11e8-93ec-a679b0374b7a.png)

**Comment by dean-rahman** [comment]
I get "TypeError: unhashable type: 'list'"

**Comment by SiyaoCai** [comment]
> I get "TypeError: unhashable type: 'list'"

Check the function name, Change it to ox.plot_graph_routes.

**Comment by bpben** [comment]
Hi, this is sort of related, but I think maybe something changed about the methods for specifying route color in a recent version? It used to be that I could specify different colors for each edge of the route, but now it doesn't seem to accept that.  I think this question was trying to do something similar, but it seems like the solution just gives you a method for specifying the color of the entire route.  I can open up a separate issue, but maybe I'm just not understanding something.

Previously my code was:
```
import matplotlib
cmap = matplotlib.cm.get_cmap('coolwarm')
rc = [cmap(x) for p in v for x in p]
ox.plot_graph_routes(Graph, routes, route_colors=rc)
```

This used to work, now it seems to expect a single element per route (e.g. "red").

**Comment by gboeing** [comment]
@bpben I'm not clear what your code is intended to do, because `p` is undefined in `rc = [cmap(x) for p in v for x in p]`. Note that `plot_graph_routes` was improved to make this easier several versions ago. See [the docs](https://osmnx.readthedocs.io/en/stable/osmnx.html#osmnx.plot.plot_graph_routes) for how to use this function to plot multiple routes with their own colors. For instance:

```python
routes = [route1, route2, route3]
rc = ['r', 'y', 'c']
fig, ax = ox.plot_graph_routes(G, routes, route_colors=rc, route_linewidth=6, node_size=0)
```

See also https://stackoverflow.com/a/53708286/7321942

**Comment by bpben** [comment]
Thanks for the quick response.  Sorry, I edited to fix: `rc` just stores the color for each individual edge within any route.  The solutions here seem to give one color for the entire route.  I'm trying to give a color for each edge in a route.  This used to work with the code above, but it no longer does.  Would it be easier if I just made a new issue here? I can try and put together a minimal example of this.
